### PR TITLE
NEXUS-8418 Flatten NuGet Routes

### PR DESCRIPTION
**Pull request by mrprescott**
https://issues.sonatype.org/browse/NEXUS-8418

### CODE REVIEW

**Comment by kellyrob99** [comment]
+1
Should probably update the readme examples to remove the extra 'upload' in the url

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/matchers/ActionMatcher.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ */
+public class ActionMatcher
+    extends ComponentSupport
+    implements Matcher
+{
+  private final List<String> allowedActions;
+
+  public ActionMatcher(final String... allowedMethods) {
+    checkNotNull(allowedMethods);
+    checkArgument(allowedMethods.length > 0, "at least one allowed method must be specified");
```
Comment (jdillon) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/matchers/ActionMatcher.java]: action/method name mismatch, pick one

**Comment by mrprescott** [comment]
Indeed - fixed terminology, README.

**Comment by mrprescott** [comment]
http://bamboo.s/browse/NX3-OSSF177-1 :white_check_mark:

**Comment by cstamas** [comment]
+1

**Comment by mrprescott** [comment]
Manually merged as c7dfdd9

**Pull request closed without merging**